iidm/iidm-api/src/main/java/com/powsybl/iidm/network/VariantManager.java

```diff
@@ -56,6 +56,20 @@
      */
     void cloneVariant(String sourceVariantId, List<String> targetVariantIds);
 
+    /**
+     * Create or overwrite a variant by cloning an existing one.
+     *
+     * @param sourceVariantId the source variant id
+     * @param targetVariantIds the target variant id list (the ones that will be created)
+     * @throws com.powsybl.commons.PowsyblException
+     *                         if the source variant is not found or if a variant with
+     *                         an id of targetStateIds already exists
+     */
+    default void cloneVariant(String sourceVariantId, List<String> targetVariantIds, boolean mayOverwrite) {
```

Comment:
Do we really want to keep the old method that throws an exception?

The javadoc is not OK: you forget to explain the third parameter. This method throws an exception only if the mayOverwrite is equal to false. Maybe the default implementation is not correct. Maybe we should add a try/catch to respect the contract of this method.

Reply:
I think we should either have no default implementation or throw an `UnsupportedOperationException` :
in any case the default implementation will not respect the contract (if you catch the exception, the variant is not copied anyway).